pkg/proxy/userspace/proxysocket.go

```diff
@@ -99,8 +99,11 @@ func tryConnect(service proxy.ServicePortName, srcAddr net.Addr, protocol string
 		outConn, err := net.DialTimeout(protocol, endpoint, retryTimeout*time.Second)
 		if err != nil {
 			if isTooManyFDsError(err) {
+				proxier.recorder.Eventf(proxier.nodeRef, "TooManyFDsError", "Dial failed, panicking")
+				time.Sleep(100 * time.Millisecond)
 				panic("Dial failed: " + err.Error())
 			}
+			proxier.recorder.Eventf(proxier.nodeRef, "DialFailed", "Dial failed")
```

Comment:
like elsewhere, this really should be annotated to the Service or Endpoints object, not the node.  And it should include the endpoint.